sopel/config/types.py

```diff
@@ -276,33 +290,64 @@ def parse(self, value):
             multi-line string.
         """
         if "\n" in value:
-            items = [
+            items = (
                 # remove trailing comma
                 # because `value,\nother` is valid in Sopel 7.x
                 item.strip(self.DELIMITER).strip()
-                for item in value.splitlines()]
+                for item in value.splitlines())
         else:
             # this behavior will be:
             # - Discouraged in Sopel 7.x (in the documentation)
             # - Deprecated in Sopel 8.x
             # - Removed from Sopel 9.x
             items = value.split(self.DELIMITER)
 
-        value = list(filter(None, items))
-        if self.strip:  # deprecate strip option in Sopel 8.x
-            return [v.strip() for v in value]
-        else:
-            return value
+        items = (self.parse_item(item) for item in items if item)
+        if self.strip:
+            return [item.strip() for item in items]
+
+        return list(items)
+
+    def parse_item(self, item):
+        """Parse one ``item`` from the list.
+
+        :param str item: one item from the list to parse
+        :rtype: str
+
+        If ``item`` matches the :attr:`QUOTE_REGEX` pattern, then it will be
+        unquoted. Otherwise it's returned as it is.
```

Comment:
```suggestion
        unquoted. Otherwise it's returned as-is.
```

Maybe this is a colloquialism? 🤔

Reply:
No idea, my English is tainted by school, and we didn't learn about urban dictionary.

Reply:
Oh, it's not slang (which is what UD tends to cover). But it _is_ an idiomatic expression, which is just as likely to be left out of formal language education. 😆